public/app/features/dashboard-scene/scene/DashboardGridItem.tsx

```diff
@@ -104,6 +108,15 @@ export class DashboardGridItem extends SceneObjectBase<DashboardGridItemState> i
   }
 
   private _onVariableUpdateCompleted(): void {
+    // if there are existing repeated panels on status loading,
+    // we need to reset the prevRepeatValues to undefined to trigger the repeat again
+    if (this.state.repeatedPanels) {
+      const data = sceneGraph.getData(this.state.repeatedPanels[0]);
+      if (data.state.data?.state === LoadingState.Loading) {
+        this._prevRepeatValues = undefined;
+      }
+    }
+
```

Comment:
can you describe the scenario here? if the current variable values are the same as _prevRepeatValues why do we need to redo the repeats?

Reply:
think I understand (by watching the video) but this feels like a the wrong way to fix it, the variable does not look to change in the video, so why should we redo the repeats? feels like it's the loading the state of the panels that is the bug

Reply:
We need to redo the repeats in case the vizPanel state changes, in this case it is in a loading state, but since we have `prevRepeatValues`, `performRepeat` will return early and not update vizPanels in a done state

Reply:
> think I understand (by watching the video) but this feels like a the wrong way to fix it, the variable does not look to change in the video, so why should we redo the repeats? feels like it's the loading the state of the panels that is the bug

Yeah, I'm not very happy myself with this. Not sure how to tackle it in a better way. The panel state gets updated properly but since `performRepeat` returns early we never update the repeated panels with the new state.